### PR TITLE
Narrow the set of cases when we pop up start page after extension upgrade

### PR DESCRIPTION
**Pull request by karolz-ms**
Fixes [issue 2953](https://github.com/microsoft/vscode-docker/issues/2953)

### CODE REVIEW

File: src/scaffolding/wizard/OpenStartPageStep.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { AzureWizardExecuteStep } from "vscode-azureextensionui";
+import { openStartPageAfterExtensionUpdate } from "../../commands/startPage/openStartPage";
+import { ScaffoldingWizardContext } from "./ScaffoldingWizardContext";
+
+export class OpenStartPageStep extends AzureWizardExecuteStep<ScaffoldingWizardContext> {
+    public constructor(public readonly priority: number) {
+        super();
+    }
+
+    // eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (karolz-ms) [inline src/scaffolding/wizard/OpenStartPageStep.ts:15]: I am not entirely sure what is the best practice for unused function arguments...

Comment (philliphoff) [reply src/scaffolding/wizard/OpenStartPageStep.ts:15]: If you don't use *any* of the arguments, you can simply omit them.  JS passes all arguments given it by the caller to the called function regardless of whether the function actually declares/uses them.

File: src/scaffolding/wizard/OpenStartPageStep.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardExecuteStep } from "vscode-azureextensionui";
+import { openStartPageAfterExtensionUpdate } from "../../commands/startPage/openStartPage";
+import { ScaffoldingWizardContext } from "./ScaffoldingWizardContext";
+
+export class OpenStartPageStep extends AzureWizardExecuteStep<ScaffoldingWizardContext> {
+    public constructor(public readonly priority: number) {
```
Comment (philliphoff) [inline src/scaffolding/wizard/OpenStartPageStep.ts:11]: Is `priority` used for anything?

**Review by philliphoff — APPROVED** [review]

File: src/scaffolding/wizard/OpenStartPageStep.ts (hunk 2 above)
Comment (karolz-ms) [reply src/scaffolding/wizard/OpenStartPageStep.ts:11]: It is used by the wizard framework to order the steps, and is an abstract member, so TS is unhappy if it is not there.

**Pull request merged**